### PR TITLE
Load urls from file

### LINKED ISSUE #12: Load URLs from a file
A plain text file with a list of URLs.

### PR DESCRIPTION
**Pull request by nappex**
fix #12

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -24,3 +24,5 @@ onigumo-*.tar
 
 # Temporary files, for example, from tests.
 /tmp/
+
+onigumo
```
Comment (Glutexo) [inline .gitignore:28]: This is not relevant to the issue. If it was, it’d be nice to attach a comment that this is a compiled binary.

Comment (Glutexo) [inline .gitignore:28]:
```suggestion
```

File: lib/onigumo.ex [hunk 2]
```diff
@@ -2,19 +2,29 @@ defmodule Onigumo do
...
       body: body
-    } = http.get!(@url)
+    } = http_client.get!(url)
+
+    File.write!(@output_filename, body)
+  end
 
-    File.write!(@filename, body)
+  def load_urls(filepath) do
+    File.stream!(filepath, [:read], :line)
```
Comment (Glutexo) [inline lib/onigumo.ex:26]: The `:read` and `:line` part is redundant, but it makes it nicely explicit that we are **reading** the file by **lines**.

**Review by Glutexo — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (nappex) [reply .gitignore]: OK

File: lib/onigumo.ex (hunk 2 above)
Comment (nappex) [reply lib/onigumo.ex:26]: good point

File: test/onigumo_test.exs [hunk 3]
```diff
@@ -2,23 +2,37 @@ defmodule OnigumoTest do
...
-    assert("hello\n" = File.read!(@filename))
+    assert(:ok == Onigumo.download(HTTPoisonMock, @url))
+    assert("Body from: #{@url}" == File.read!(@filename))
   end
+
+
+  @tag :tmp_dir
+  test("load URL from file", %{tmp_dir: tmp_dir}) do
+    filepath = Path.join(tmp_dir, @testfile_with_urls)
+    content = @url <> " \n"
```
Comment (Glutexo) [inline test/onigumo_test.exs:31]:
```suggestion
    content = "#{@url}\n"
```

**Review by Glutexo — APPROVED** [review]

File: test/onigumo_test.exs (hunk 3 above)
Comment (nappex) [reply test/onigumo_test.exs:31]: :+1:

**Review by Glutexo — COMMENTED** [review]
I have a pending review here. Probably not relevant any more. Submitting just to clean up.

**Comment by Glutexo** [comment]
Marked everything as resolved.

**Pull request merged**